test/test_pointevaluation.jl

```diff
@@ -35,18 +43,59 @@ function scalar_field()
     # @test f.(points) ≈ vals
 end
 
-function vector_field()
+function test_pe_embedded()
+    mesh = generate_grid(QuadraticQuadrilateral, (3, 3))
+    perturbate_standard_grid!(mesh, 1/10)
+    mesh = Grid(mesh.cells, map(x->Node(Vec((x.x[1], x.x[2], x.x[1]+x.x[2]))), mesh.nodes))
+
+    f(x) = x[1]^2+x[2]^2
+
+    ip_f = Lagrange{RefQuadrilateral,2}() # function interpolation
+    ip_g = Lagrange{RefQuadrilateral,2}()^3 # geometry interpolation
+
+    # points where we want to retrieve field values
+    points = Vec{3,Float64}[]
+
+    # compute values in quadrature points
+    qr = QuadratureRule{RefQuadrilateral}(3) # exactly integrate quadratic field
+    cv = CellValues(qr, ip_f, ip_g)
+    qp_vals = [Vector{Float64}(undef, getnquadpoints(cv)) for _ in 1:getncells(mesh)]
+    for cellid in eachindex(mesh.cells)
+        xe = getcoordinates(mesh, cellid)
+        reinit!(cv, xe)
+        for qp in 1:getnquadpoints(cv)
+            x = spatial_coordinate(cv, qp, xe)
+            qp_vals[cellid][qp] = f(x)
+            push!(points, x)
+        end
+    end
+
+    # do a L2Projection for getting values in dofs
+    # @test_throws MethodError projector = L2Projector(ip_f, mesh; geom_ip=ip_g)
+    projector = L2Projector(ip_f, mesh)
+    projector_vals = project(projector, qp_vals, qr)
+
+    # set up PointEvalHandler and retrieve values
+    ph = PointEvalHandler(mesh, points)
+    @test all(x -> x !== nothing, ph.cells)
+
+    # vals = evaluate_at_points(ph, projector, projector_vals)
+    # @test f.(points) ≈ vals
```

Comment:
Why the commented out tests?

Reply:
They did not work. I fixed them with the last commit.

Reply:
OK, maybe then a comment regarding the exact integration of a linear function `f(x)` for the quadratic interpolation with the chosen QR?

Reply:
What do you mean? We do not really integrate anything in the test itself, but use a setup which would be exact if we solve the problem and test if we can evaluate some function at the quadrature points with high precision.

Reply:
Ah, on second thought, no I don't understand why you needed to make `f(x)` linear to make the test work then. Because the order shouldn't add up, we just need an interpolation which can represent `f(x)` exactly, so a quadratic `f(x)` should work?

The L2Projector does the integration I guess, so that's where you need sufficient accuracy to capture the nonlinearities in the test, and for that it should suffice to integrate the interpolation exactly...

Reply:
> Ah, on second thought, no I don't understand why you needed to make `f(x)` linear to make the test work then. Because the order shouldn't add up, we just need an interpolation which can represent `f(x)` exactly, so a quadratic `f(x)` should work?
> 
Quadratic f also works, but will be associated with larger error for points outside the ball of radius 1, because there are tiny errors in the found position which will be amplified.


> The L2Projector does the integration, so that's where you need sufficient accuracy to capture the nonlinearities in the test, and for that it should suffice to integrate the interpolation exactly...

Oh, right. :D